targets/photos/services/onPhotoUpload.js

```diff
@@ -29,93 +28,112 @@ import {
 } from 'photos/ducks/clustering/albums'
 import { prepareDataset } from 'photos/ducks/clustering/utils'
 
-// Retrieve the parameters used to compute the clustering
-const clusteringParameters = (dataset, setting) => {
-  const params = defaultParameters(setting)
-  if (!params) {
-    return null
-  }
-  if (!params.epsTemporal) {
-    params.epsTemporal = computeEpsTemporal(dataset, PERCENTILE)
-  }
-  if (!params.epsSpatial) {
-    params.epsSpatial = computeEpsSpatial(dataset, PERCENTILE)
-  }
-  const epsMax = Math.max(params.epsTemporal, params.epsSpatial)
-  if (!params.maxBound) {
-    params.maxBound =
-      epsMax * 2 < DEFAULT_MAX_BOUND ? epsMax * 2 : DEFAULT_MAX_BOUND
-  }
-  if (!params.cosAngle) {
-    params.cosAngle = gradientAngle(epsMax, COARSE_COEFFICIENT)
-  }
-  return params
-}
-
 // Compute the actual clustering based on the new dataset and the existing albums
-const computeClusters = async (dataset, albums, params) => {
-  if (albums && albums.length > 0) {
-    const clusterize = await albumsToClusterize(dataset, albums)
-    if (clusterize) {
-      for (const [id, photos] of Object.entries(clusterize)) {
-        // TODO adapt params to the period
-        const reachs = reachabilities(photos, spatioTemporalScaled, params)
-        const clusters = gradientClustering(photos, reachs, params)
-        if (clusters.length > 0) {
-          const ids = id.split(':')
-          const albumsToSave = findAlbumsByIds(albums, ids)
-          saveClustering(clusters, albumsToSave)
-        }
+const createNewClusters = async (setting, dataset, albums) => {
+  const clusterize = await albumsToClusterize(dataset, albums)
+  if (clusterize) {
+    for (const [id, photos] of Object.entries(clusterize)) {
+      const params = findPhotosDefaultParameters(setting, photos)
+      if (!params) {
+        log('warn', 'No parameters for clustering found')
+        break
```

Comment:
Just to be sure, if we can't find the params for one cluster, we stop doing *all* the clusters? Because otherwise we'd want `continue` instead of `break` here :)

Reply:
Right, well seen!